pra_request_tracker/pra_request_tracker/record_requests/models.py

```diff
@@ -75,21 +76,27 @@ def status_label(self):
     def files(self):
         return self.recordrequestfile_set.all()
 
+    @cached_property
+    def correspondences(self):
+        return list(self.correspondence_set.prefetch_related("recordrequestfile_set"))
+
 
 class Correspondence(BaseModel):
     request = models.ForeignKey(RecordRequest, on_delete=models.CASCADE)
-    from_ = models.CharField(max_length=256, name="from")
-    to = models.CharField(max_length=256)
+    from_address = models.CharField(max_length=256)
+    to_address = models.CharField(max_length=256)
     subject = models.CharField(max_length=256)
     body = models.TextField()
-    date = models.DateTimeField()
+    date = models.DateTimeField(blank=True)
 
     def __str__(self):
         return f"Correspondence({self.pk}) {self.subject} for {self.request}"
 
-    @property
-    def fromm(self):
-        return getattr(self, "from")
+    def save(self, *args, **kwargs):
+        if not self.date:
```

Comment:
I think we can set this as a default instead of doing it on save? https://docs.djangoproject.com/en/3.2/ref/models/fields/#django.db.models.DateField.auto_now_add (see the note right after this section)

Reply:
I tried using `auto_now_add` but it also sets `editable=False`: https://github.com/django/django/blob/00ea883ef56fb5e092cbe4a6f7ff2e7470886ac4/django/db/models/fields/__init__.py#L1151

This makes it impossible to set a custom date, in the case of back porting correspondences for example.

Reply:
Oh sorry, I specifically meant this bit!
> If you want to be able to modify this field, set the following instead of auto_now_add=True:
> For DateField: default=date.today - from datetime.date.today()
> For DateTimeField: default=timezone.now - from django.utils.timezone.now()

Reply:
Oh nice!

Reply:
Done! Thanks for the tip.